python/lsst/ts/externalscripts/auxtel/calsys_take_narrowband_data.py

```diff
@@ -99,39 +188,42 @@ def __init__(self, index):
         self.log.info("Configure started")
 
         nelt = 1
-        kwargs = locals()
         for argname in ("wavelengths", "integration_times", "mono_grating_types",
                         "mono_entrance_slit_widths", "mono_exit_slit_widths",
                         "image_types", "lamps", "fiber_spectrometer_delays",
                         "latiss_filter", "latiss_grating", "latiss_stage_pos",
-                        "nimages_per_wavelength", "shutter", "image_sequence_name","fiber_spectrograph_integration_times"):
-            value = kwargs[argname]
+                        "nimages_per_wavelength", "shutter",
+                        "image_sequence_name", "fiber_spectrograph_integration_times"):
+            value = getattr(config, argname)
             if is_sequence(value):
                 nelt = len(value)
                 break
-        self.file_location=os.path.expanduser(file_location)
-        self.setup_spectrograph = setup_spectrograph
-        self.take_image = take_image
-        self.script_type = script_type
+        self.file_location = os.path.expanduser(self.config.file_location)
+        self.setup_spectrograph = self.config.setup_spectrograph
+        self.take_image = self.config.take_image
+        self.script_type = self.config.script_type
```

Comment:
I very strongly recommend *not* copying data from config to separate fields in the script. Just use `self.config.script_type` when you want the script type. Copying adds a lot of boilerplate and is error-prone. Note that the schema constrains the data types in the config much as a class definition would, so it's not as if you'll be getting arbitrary nonsense data that you have to validate.